lte/protos/session_manager.proto

```diff
@@ -540,3 +556,251 @@ service CentralSessionController {
+  SubscriberID sid = 25;
+  string imei = 8; // pei - permanent equipment ID. Variable length of 16 bytes. TS-23.003 - 6.4 Like IMEI or IMEISV
+  string gpsi = 9; //Lenght is 10 bytes + 2 bytes country code generic public subscription id.
+  string user_location_info = 10; //up to 6 byte. TS-23.003 - 28.15.3
+  string pcf_id = 11; //length 16 bytes as UUID "Universally Unique IDentifier" TS-29.571 to ietf rfc 4122
+  //Variable length of 20 bytes. TS-23.003 - 28.8 & TS-23.501 5.9.8
+  repeated LadnServiceArea ladn_service_area = 12; //Length is 10 bytes, like PLMN
+  bool pdu_session_status_notification_sub = 13;   //Subscription For PDU Session Status Notification true or false
+  DnnSelectionMode dnn_selection_mode = 14;       // TS-29-502 - 6.1.6.3.10
+  string trace_requirment = 15;                   // TS-29.571 5.6.
+  bytes integrity_protection_max_data_rate = 16;  //Length is 3 bytes, TS-24.501 9.11.4.7
+  PduSessionType pdu_session_type = 17;
+  SscMode ssc_mode = 18;
+  M5GSMCapability m5g_sm_capability = 19;  // max size go up to 15 bytes. TLV and 9.11.4.1. Flattend
+  uint32 max_pkt_filter_supported = 20;    //Maximum no. of supported packet filters
+  bool pdu_session_req_always_on = 21;     //Always-on PDU session requested TS-24.501 9.11.4.4
+  string sm_pdu_dn_req = 22;               //SM PDU DN request container. Max size up to 255. TLV and 9.11.4.16.
+  M5GSMCause m5gsm_cause = 23;
+  RequestedQosRules requested_qos_rules = 24; //TS-24501 - 9.11.4.13 minimum 7 and max 65538, variable length.  Flattend .
+}
+
+message SetSMSessionContext {
+  CommonSessionContext common_context = 1;  //Common message for 4g, 5g and WiFi.
+  RatSpecificContext   rat_specific_context = 2;
+}
+message SmContextVoid {
+
+  string output = 1;
+
+}
+
+
+//SET API for 3 procedures request from AMF to SMF
+        service AmfPduSessionSmContext {
+                rpc SetAmfSessionContext (SetSMSessionContext) returns (SmContextVoid);
+        }
+
+////////// SET MESSAGE PROTO DEFINATION OF SMF TO AMF ON RESPONSE /////////////
+
+//4G response message has not been refactored yet. Hence only considering M5G.
+message RatSpecificContextAccess {
+  //oneof context {
+    //LTESessionContextAccess lte_context = 1;
+    M5GSMSessionContextAccess m5g_session_context_rsp = 2;
+  //}
+}
+
+//AuthorizedQosRules
+//The purpose of the QoS rules information element is to indicate a set of QoS rules to be used by the UE, 
+//where each QoS rule is a set of parameters.
+//a) for classification and marking of uplink user traffic; and
+//b) for identification of a QoS flow which the network is to use for a particular downlink user traffic.
+//As the UE needs to be aware of a QoS flow which the network is to use for a particular downlink user traffic
+//e.g. to determine whether a resource is available for downlink media of a media stream of an SDP media
+//description provided by the UE in an IMS session.
+
+message AuthorizedQosRules {	
```

Comment:
We can collapse both `AuthorizedQosRule` and `RequestedQosRules` into the same structure, it seems.

Reply:
Responded in previous comment.

Reply:
Adapted as QoSRules. Described in above comments.